File: pset6/readability/readability.py
from cs50 import get_string

paragraph = get_string("Text: ")
    
# count words
word_count = paragraph.split()
W = len(word_count)

# count letters
# count sentence 
letter_count = []
L = 0
S = 0

length = len(paragraph)

for i in range(length):
    letter_count.append(ord(paragraph[i]))

for i in range(len(letter_count)):
    if letter_count[i] > 64 and letter_count[i] < 90:  # use ascii code
        L += 1
    elif letter_count[i] > 96 and letter_count[i] < 123:
        L += 1
    elif letter_count[i] == 33 or letter_count[i] == 46 or letter_count[i] == 63:
        S += 1

    
# use formula 
avg_L = L*100/W
avg_S = S*100/W

grade = round(0.0588 * avg_L - 0.296 * avg_S - 15.8)

# print result 
if grade < 1:
    print("Before Grade 1")
elif grade > 16:
    print("Grade 16+")
else:
    print("Grade", grade)
   

File: pset2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>


int main(void)
{
    string text = get_string("text: ");

    int L = 0;

    for (int i = 0, m = strlen(text); i < m; i++) //counter +1 if it detects letters
    {
        if (isalpha(text[i]))
        {
            L = L + 1;
        }
    }

    printf("%i\n", L);

    int S = 0;

    for (int j = 0, n = strlen(text); j < n; j++) 
    {
        if ((text[j]) == 33 || (text[j]) == 46 || (text[j]) == 63)

        {
            S = S + 1;
        }
    }

    printf("%i\n", S);
    int W = 1;
    for (int k = 0, p = strlen(text); k < p; k++)
    {
        if (isspace(text[k]))

        {
            W = W + 1;
        }
    }

    printf("%i\n", W);

    int grade = round(0.0588 * L * 100 / W - 0.296 * S * 100 / W - 15.8);

    if (grade < 1)  //print the result
    {
        printf("Before Grade 1\n");
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }

}




